src/charm.py

```diff
@@ -244,6 +281,11 @@ def get_ovn_node_ips(self):
         ]
         return node_ips
 
+    def handle_endpoints_changed(self, _):
+        if self.remote_write_consumer.endpoints and self.unit.is_leader():
+            # Get the last available endpoint reported in the relation data.
+            self.apply_grafana_agent(self.remote_write_consumer.endpoints[-1]["url"])
```

Comment:
OK, so if there's any endpoint at all, we just use the `url` field from the last one.  Is there anything special about the last one that different from the others?  Is there a need to wait until all the `url`'s match?

Reply:
From the Prometheus [docs](https://prometheus.io/docs/prometheus/latest/configuration/configuration/#remote_write) and Grafana Agent [docs](https://grafana.com/docs/agent/v0.26/configuration/create-config-file/), I *gather* that Prometheus can only use one remote write endpoint. It might be worth testing whether it is possible to pass multiple endpoints to the agent configuration.